src/ast.c

```diff
@@ -267,6 +268,7 @@ static jl_ast_context_t *jl_ast_ctx_enter(void)
 
 static void jl_ast_ctx_leave(jl_ast_context_t *ctx)
 {
+    JL_SIGATOMIC_END();
```

Comment:
And in case you need it now, I think making jl_sigint_safepoint noop should work around it for now